### PR TITLE
hibernate-1

### PR DESCRIPTION
**Pull request by entertainment-veks**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,27 +60,14 @@
...
-                    <configuration>
-                        <source>${jdk.version}</source>
-                        <target>${jdk.version}</target>
-                        <encoding>${project.build.sourceEncoding}</encoding>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
     </build>
 </project>
```
Comment (Fedich09) [inline pom.xml:73]:
```suggestion
</project>
```
add empty line

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Movie entity", e);
```
Comment (Fedich09) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:27]:
```suggestion
            throw new RuntimeException("Can't insert Movie entity", e);
```
throw your custom exception

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaQuery<Movie> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Movie.class);
+            criteriaQuery.from(Movie.class);
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all", e);
```
Comment (Fedich09) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:39]:
```suggestion
            throw new RuntimeException("Can't get all", e);
```
throw your custom exception

File: src/main/resources/hibernate.cfg.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <property name="connection.url">jdbc:mysql://localhost:3306/hibernate?serverTimezone=UTC</property>
+        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="connection.username">root</property>
+        <property name="connection.password">1234</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="hibernate.model.Movie"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (Fedich09) [inline src/main/resources/hibernate.cfg.xml:18]:
```suggestion
</hibernate-configuration>
```
add empty line

**Review by Fedich09 — CHANGES_REQUESTED** [review]

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (Sofasmile) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:18]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:24]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new CustomDaoException("Can't insert Movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:29]:
```suggestion
            throw new CustomDaoException("Can't insert Movie entity " + movie, e);
```

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        } finally {
+            session.close();
```
Comment (Sofasmile) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:31]: let's add check for null value

File: src/main/java/hibernate/model/CinemaHall.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
+package hibernate.model;
+
+import java.util.Objects;
+
+public class CinemaHall {
```
Comment (Sofasmile) [inline src/main/java/hibernate/model/CinemaHall.java:5]: this class shouldn't be in the PR

you car remove all redundant files from git control using the command: `git rm --cached filename`
after that, you should commit all changes and push them to the PR.

Then the files will stay locally in the project, but they will not be in PR

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**